plugins/fileservices/fileservices.cpp

```diff
@@ -1611,10 +1611,14 @@ FILESERVICES_API void FILESERVICES_CALL fslFinishSuperFileTransaction(ICodeConte
 }
 
 
-FILESERVICES_API char *  FILESERVICES_CALL fsForeignLogicalFileName(ICodeContext *ctx, const char *_lfn,const char *foreigndali,bool abspath)
+static char * implementForeignLogicalFileName(ICodeContext *ctx, const char *_lfn, const char *foreigndali, bool abspath, bool omitClusterPrefix)
 {
     StringBuffer lfns;
-    constructLogicalName(ctx, _lfn, lfns);
+
+    Owned<IConstWorkUnit> wu;
+    if (!omitClusterPrefix && _lfn&&(*_lfn != '~'))
+        wu.setown(getWorkunit(ctx));
+    constructLogicalName(wu, _lfn, lfns);
```

Comment:
Can wu end up NULL (in Roxie, for example)?

Reply:
it's okay for wu to be null, code in constructLogicalName(IConstWorkUnit *,..
will check for null.

NB: in this regard, the logic is the same as before this PR, i.e. it used to unconditionally get a 'wu' (if not absolute prefix), and then call on as it does now. 'wu' could still be null.